src/arch/loongarch64.cpp

```diff
@@ -0,0 +1,94 @@
+#include "arch.h"
+
+#include <algorithm>
+#include <array>
+
+// SP points to the first argument that is passed on the stack
+#define ARG0_STACK 0
+
+namespace bpftrace {
+namespace arch {
+
+// clang-format off
+static std::array<std::string, 17> registers = {
```

Comment:
I have an unfinished branch at https://github.com/iovisor/bpftrace/compare/master...chenhengqi:bpftrace:support-loongarch

Did you really test this change ?

Reply:
I've done compile verification locally

Reply:
> I have an unfinished branch at [master...chenhengqi:bpftrace:support-loongarch](https://github.com/iovisor/bpftrace/compare/master...chenhengqi:bpftrace:support-loongarch)
> 
> Did you really test this change ?

I've done compile verification locally

Reply:
Just opened #2466, your change does NOT work actually.

Reply:
> Just opened #2466, your change does NOT work actually.

Why didn't my changes actually work

Reply:
Your implementation does not comply with the [ABI specification](https://loongson.github.io/LoongArch-Documentation/LoongArch-ELF-ABI-EN.html). For example, it says that there are 32 general-purpose registers while you only state 17, etc.

#2466 looks much better on that front, so I'd favor that one over this PR.